test/robot/web/www/controllers/actionController.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var newActionController = function (commitState, defaultPopFunction) {
+  var controller = {
+    view: newBreadcrumbView(),
+    changing: false,
+    targetHash: "",
+    actionList: [],
+    pushAction: function (actionHash, popFunction, forcedHash) {
+      var action = {
+        hash: actionHash,
+        popStack: [popFunction],
+      };
+      action.index = controller.actionList.push(action);
+      var fullHash;
```

Comment:
[You should declare all vars at the top of the function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting). Here and elsewhere.